spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java

```diff
@@ -198,7 +198,7 @@ public AuditRecordController auditController(
 	}
 
 	@Bean
-	@ConditionalOnBean({ StreamDefinitionRepository.class, StreamDeploymentRepository.class })
+	@ConditionalOnBean({ StreamDefinitionRepository.class })
```

Comment:
The reason we enforced the `StreamDeploymentRepository` bean in this was not to make sure both the repositories exist for all the Stream operations. Perhaps we can add a custom condition to do such thing here?

Reply:
Actually we didn't. In boot 1.x `ConditionalOnBean` is logical OR, in boot 2 it's logical AND. As `StreamDeploymentRepository` wasn't created in skipper mode, this broke. See https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide